examples/AzureSDKDemoSwiftUI/AzureSDKDemoSwiftUI/Pages/BlobListObservable.swift

```diff
@@ -0,0 +1,114 @@
+// --------------------------------------------------------------------------
+//
+// Copyright (c) Microsoft Corporation. All rights reserved.
+//
+// The MIT License (MIT)
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the ""Software""), to
+// deal in the Software without restriction, including without limitation the
+// rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+// sell copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+// IN THE SOFTWARE.
+//
+// --------------------------------------------------------------------------
+
+import Foundation
+
+import AzureCore
+import AzureIdentity
+import AzureStorageBlob
+import MSAL
+
+class BlobListViewModel {
```

Comment:
Does this need to be a class? Does it work as a struct?

Reply:
At the time being it is a class cause it made more sense for performance. 
The `downloadProgress` is called frequently and can hold onto state in the transfers dictionary. 
When I change this to a pure swift implementation I might be able to make this a struct using @Publish vars 
It will be on my radar

Reply:
Tried a couple of things but looks like the best way is to use a class. Struct won't  have as nice a design  to easily access the blob transfers progress. Lets talk about if there is a cleaner way to implement it